java/client/src/main/java/glide/api/models/configuration/BaseSubscriptionConfiguration.java

```diff
@@ -0,0 +1,96 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.configuration;
+
+import glide.api.models.configuration.ClusterSubscriptionConfiguration.PubSubClusterChannelMode;
+import glide.api.models.configuration.StandaloneSubscriptionConfiguration.PubSubChannelMode;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Client subscription configuration. Could be either {@link StandaloneSubscriptionConfiguration} or
+ * {@link ClusterSubscriptionConfiguration}.
+ */
+@Getter
+@RequiredArgsConstructor
+public abstract class BaseSubscriptionConfiguration {
+
+    /**
+     * A channel subscription mode. Could be either {@link PubSubChannelMode} or {@link
+     * PubSubClusterChannelMode}.
+     */
+    public interface ChannelMode {}
+
+    /**
+     * Callback is called for every incoming message. The arguments are:
+     *
+     * <ol>
+     *   <li>A received message
+     *   <li>User-defined context
+     * </ol>
+     */
+    // TODO change message type
+    public interface MessageCallback extends BiConsumer<Object, Object> {}
+
+    /**
+     * Optional callback to accept the incoming messages. See {@link MessageCallback}.<br>
+     * If not set, messages will be available via TODO method.<br>
+     */
+    protected final Optional<MessageCallback> callback;
```

Comment:
feels like callback subscription configuration and queued configuration could be separate objects

Reply:
How do you see that?
Is it good enough?
```java
        var regularClient =
                RedisClient.CreateClient(
                                RedisClientConfiguration.builder()
                                        .address(NodeAddress.builder().port(6379).build())
                                        .requestTimeout(3000)
                                        .subscriptionConfiguration(
                                                StandaloneSubscriptionConfiguration.builder()
                                                        .subscription(PubSubChannelMode.EXACT, "ch1")
                                                        .subscription(PubSubChannelMode.PATTERN, "chat*")
                                                                        //   from there
                                                        .build())
                                        .subscriptionCallback(callback) // I moved this
                                        .build())
                        .get();
```